WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift

```diff
@@ -2498,63 +2555,30 @@ extension AztecPostViewController {
     }
 
     fileprivate func insertDeviceImage(phAsset: PHAsset) {
-        let attachment = richTextView.replaceWithImage(at: self.richTextView.selectedRange, sourceURL: URL(string:"placeholder://")!, placeHolderImage:
-                Assets.defaultMissingImage)
+        guard let attachment = attachmentWithPlaceholder(isImage: true) as? ImageAttachment else { return }
 
         let mediaService = MediaService(managedObjectContext:ContextManager.sharedInstance().mainContext)
-        mediaService.createMedia(with: phAsset, forPost: post.objectID, thumbnailCallback: { [weak self](thumbnailURL) in
-            guard let `self` = self else {
-                return
-            }
-            DispatchQueue.main.async {
-                attachment.updateURL(thumbnailURL)
-                self.richTextView.refresh(attachment)
-            }
-        }, completion: { [weak self](media, error) in
-            guard let strongSelf = self else {
-                return
-            }
-            guard let media = media, error == nil else {
-                DispatchQueue.main.async {
-                    strongSelf.handleError(error as NSError?, onAttachment: attachment)
-                }
-                return
-            }
-
-            WPAppAnalytics.track(.editorAddedPhotoViaLocalLibrary, withProperties: WPAppAnalytics.properties(for: media, mediaOrigin: strongSelf.selectedMediaOrigin), with: strongSelf.post.blog)
-
-            strongSelf.upload(media: media, mediaID: attachment.identifier)
+        mediaService.createMedia(with: phAsset,
+                                 forPost: post.objectID,
+                                 thumbnailCallback: { [weak self](thumbnailURL) in
+                                    self?.handleThumbnailURL(thumbnailURL, attachment: attachment)
+            },
+                                 completion: { [weak self](media, error) in
+                                    self?.handleNewMedia(media, error: error, attachment: attachment, statType: .editorAddedPhotoViaLocalLibrary)
         })
     }
 
     fileprivate func insertDeviceVideo(phAsset: PHAsset) {
-        let attachment = richTextView.replaceWithVideo(at: richTextView.selectedRange, sourceURL: URL(string:"placeholder://")!, posterURL: URL(string:"placeholder://")!, placeHolderImage: Assets.defaultMissingImage)
-        attachment.progress = 0
-        richTextView.refresh(attachment)
+        guard let attachment = attachmentWithPlaceholder(isImage: false) as? VideoAttachment else { return }
```

Comment:
Please move the ` { return }` to the next line, for readability purposes:

```
guard let attachment = attachmentWithPlaceholder(isImage: false) as? VideoAttachment else { 
   return 
}
```

Reply:
@jleandroperez - I changed this line, so this is unnecessary. However, for future reference, can you tell my why you made this request? I see single line guard statements with `{ return }` throughout the app. Thanks!

Reply:
Yep, she's right, we have tons of those. I honestly prefer to have them in a new line, and with no clear readability guidelines I think it's been left to personal criteria. I'm going to make a note about this in the Tech Debt thread, for sure we have to get better at this. Thanks for pointing it out, both of you.

Reply:
To begin with, we *always* should avoid horizontal scrolling in the IDE.  `guard` statements usually have more than one condition check (and they not always just `return`, in some scenarios you might wanna call a closure, indicating a condition error). 

Even if _this_ specific case doesn't produce horizontal scrolling, for consistency's reason, it'd be great if we could have **them all** look the same way. This does not apply _just_ to this PR, there's clearly a bunch `guard else { return }` statements i did not catch, in earlier reviews 🔪 .

Thank you both :D

Reply:
Yeah, we don't really have a set guideline for it right now. Some folks prefer a new line, some folks prefer the same line. Personally I like the same line for a simple `guard` `return` that does nothing else, but I know others don't like it :)

Reply:
Thanks @jleandroperez I agree, maybe you want to propose Readability guidelines that we can discuss on and hopefully follow in the future, or we could look into adding this rule to Hound :)